components/kyma-environment-broker/common/deprovision/client.go

```diff
@@ -0,0 +1,63 @@
+package deprovision
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2/clientcredentials"
+	"net/http"
+	"time"
+)
+
+const defaultPageSize = 100
+
+// Client is the interface to interact with the KEB /deprovision API as an HTTP client using OIDC ID token in JWT format.
+type Client interface {
+	DeprovisionRuntime(params DeprovisionParameters) error
+}
+
+type DeprovisionClient struct {
+	log    logrus.FieldLogger
+	URL    string
+	client *http.Client
+}
+
+func NewDeprovisionClient(parameters DeprovisionParameters) *DeprovisionClient {
+	cfg := clientcredentials.Config{
+		ClientID:     parameters.ClientID,
+		ClientSecret: parameters.ClientSecret,
+		TokenURL:     parameters.TokenURL,
```

Comment:
should this tokenURL a absolute url?

Reply:
just checking:), what is the difference between clientcredentials.Config.Client and oauth2.NewClient? i saw others using oauth2.NewClient

Reply:
I wanted to use `clientcredentials` package, because I wanted to mimic the following command https://github.tools.sap/kyma/runtime-ops/blob/0d1ebb45b4fcb54f81367a31d6edd4176ac58da4/util.sh#L47. If I can do the same with `oauth2`, it makes sense to switch to that.

Reply:
tokenURL should be absolute URL yes. It should be provided by the kcp user in the config file ```"$HOME/.kcp/config-dev.yaml"```